packages/block-editor/src/components/block-preview/auto.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import useResizeAware from 'react-resize-aware';
+
+/**
+ * WordPress dependencies
+ */
+import { Disabled } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import BlockList from '../block-list';
+
+function AutoBlockPreview( { viewportWidth } ) {
+	const [
+		containerResizeListener,
+		{ width: containerWidth },
+	] = useResizeAware();
```

Comment:
Nice hook. Maybe `ScaledBlockPreview` should be aware or resizing, too.

Reply:
Yeah potentially. Thought, I would like to see if we could use the new preview everywhere. Maybe there are better ways to center vertically...